File: g1-gui-app/Assets/Scripts/DataTypes/SensorSample.cs
using System;
using UnityEngine;

[Serializable]
public class SensorSample
{
    public float value;
    public string sampleDateTime;

    public SensorSample Clone()
    {
        SensorSample clonedSample = new SensorSample();
        clonedSample.value = value;
        clonedSample.sampleDateTime = sampleDateTime;

        return clonedSample;
    }
}

File: forwarding-app/src/main/java/SharedDataTypes/SensorSample.java
package SharedDataTypes;

import java.time.ZonedDateTime;
import java.time.format.DateTimeFormatter;

public class SensorSample {
    public String sampleDateTime;
    public float value;

    @Override
    public String toString() {
        return "{"+String.format("%.3f", value)+" : "+sampleDateTime+"}";
    }
    public SensorSample clone(){
        SensorSample clone = new SensorSample();
        clone.sampleDateTime = sampleDateTime;
        clone.value = value;

        return clone;
    }
}
